projects/storefrontlib/src/cms-components/user/register/register.component.spec.ts

```diff
@@ -140,6 +199,29 @@ describe('RegisterComponent', () => {
         })
         .unsubscribe();
     });
+
+    it('should fetch registerUser process loading state', done => {
+      let loadingState: boolean;
+      component.ngOnInit();
+
+      // when false
+      registerUserIsLoading.next(false);
+      component.loading$.subscribe(loading => {
+        loadingState = loading;
+        done();
+      });
+
+      expect(loadingState).toEqual(false);
+
+      // when true
+      registerUserIsLoading.next(true);
+      component.loading$.subscribe(loading => {
```

Comment:
Those subscriptions and assertions with done won't work. You could test html to check if spinner is shown instead of testing implementation details, which is loading$ variable.